packages/display/src/DisplayObject.ts

```diff
@@ -714,8 +715,7 @@ export abstract class DisplayObject extends EventEmitter
         {
             this.parent.removeChild(this);
         }
-        this.emit('destroyed');
-        this.removeAllListeners();
+        this._destroyed = true;
```

Comment:
I put this after the `removeChild` because that's a common operation and not really destructive for this object, but perhaps we should be setting the flag before that anyway?